src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -338,6 +338,31 @@ public override async Task ValidateIntegrityAsync(SignatureContent signatureCont
 #endif
         }
 
+        public override string GetContentHashForSignedPackage(CancellationToken token)
+        {
+            token.ThrowIfCancellationRequested();
+
+            ThrowIfZipReadStreamIsNull();
+
+            using (var zip = new ZipArchive(ZipReadStream, ZipArchiveMode.Read, leaveOpen: true))
+            {
+                var signatureEntry = zip.GetEntry(SigningSpecifications.SignaturePath);
+
+                if (signatureEntry == null ||
+                    !string.Equals(signatureEntry.Name, SigningSpecifications.SignaturePath, StringComparison.Ordinal))
```

Comment:
can `signatureEntry.Name` ever be different than `SigningSpecifications.SignaturePath` since we are using it in `GetEntry(...)`?